src/lib/common/sol-power-supply-impl-linux.c

```diff
@@ -181,37 +181,33 @@ _get_enum_prop(const char *name, enum sol_power_supply_prop prop,
     return 0;
 }
 
-static bool
+static int
 _list_supplies_cb(void *data, const char *dir_path, struct dirent *ent)
 {
     struct list_data *ldata = data;
     enum sol_power_supply_type type;
     char *name;
     int r;
 
-    if ((!strcmp(ent->d_name, ".")) || (!strcmp(ent->d_name, ".."))) {
-        return false;
-    }
-
     if (ldata->check_type) {
         r = sol_power_supply_get_type(ent->d_name, &type);
-        SOL_INT_CHECK(r, < 0, false);
+        SOL_INT_CHECK(r, < 0, r);
```

Comment:
Here you are changing previous behaviour. What if on error the idea is really scan next file?